model/Uzivatel/Finance.php

```diff
@@ -632,88 +524,102 @@ public function maximalniPocetModrychTricekZdarma(): int
             : 0;
     }
 
-    public function muzeObjednavatModreTrickoSeSlevou(): bool
-    {
-        return $this->u->maPravo(Pravo::MUZE_OBJEDNAVAT_MODRA_TRICKA);
-    }
-
-    public function muzeObjednavatCerveneTrickoSeSlevou(): bool
-    {
-        return $this->u->maPravo(Pravo::MUZE_OBJEDNAVAT_CERVENA_TRICKA);
-    }
-
     /**
      * Viz ceník
+     * @return array<string>
      */
-    public function slevyVse()
+    public function slevyVse(): array
     {
-        return $this->cenik->slevySpecialni();
+        return $this->cenik()->slevySpecialni();
     }
 
     /**
      * Vrátí součinitel ceny aktivit, tedy slevy uživatele vztahující se k
      * aktivitám. Vrátí hodnotu.
      */
-    private function soucinitelAktivit()
+    private function soucinitelCenyAktivit(): float
     {
-        if (!isset($this->soucinitelCenyAKtivit)) {
+        if ($this->soucinitelCenyAKtivit === null) {
             // pomocné proměnné
             $sleva = 0; // v procentech
             // výpočet pravidel
             if ($this->u->maPravo(Pravo::AKTIVITY_ZDARMA)) {
-                // sleva 100%
-                $sleva                   += 100;
-                $this->slevyNaAktivity[] = 'sleva 100%';
-            } else if ($this->u->maPravo(Pravo::CASTECNA_SLEVA_NA_AKTIVITY)) {
-                // sleva 40%
-                $sleva                   += 40;
-                $this->slevyNaAktivity[] = 'sleva 40%';
+                $sleva += self::PLNA_SLEVA_PROCENT;
+            } elseif ($this->u->maPravo(Pravo::CASTECNA_SLEVA_NA_AKTIVITY)) {
+                $sleva += self::CASTECNA_SLEVA_PROCENT;
             }
-            if ($sleva > self::$maxSlevaAktivit) {
+            if ($sleva > self::MAX_SLEVA_AKTIVIT_PROCENT) {
                 // omezení výše slevy na maximální hodnotu
-                $sleva = self::$maxSlevaAktivit;
+                $sleva = self::MAX_SLEVA_AKTIVIT_PROCENT;
             }
```

Comment:
nikdy nenastane

Reply:
Je to tak. Ale nechal bych to tam jako paranoidní pojistku, kdyby někdo v budoucnu hrábnul do slev.